cmake/compiler.cmake

```diff
@@ -211,6 +211,19 @@ option(ENABLE_WERROR "Make all compiler warnings into errors" OFF)
 
 option(ENABLE_UB_SANITIZER "Make the compiler generate runtime code to perform undefined behaviour checks" OFF)
 
+set(HARDENING_FLAGS " ")
```

Comment:
I compiled tarantool using `cmake3 . -DCMAKE_BUILD_TYPE=RelWithDebInfo && make` on CentOS 7 (`packpack/packpack:el-7` docker image) and run it under GDB. I constantly get `0x41c7b0` as address of `main`. I have `set disable-randomization off` in `~/.gdbinit`. Does it mean that ASLR is not in effect or I'm doing something wrongly?

Reply:
Please try to start it and then `print $rip`.

Reply:
```c
[root@b9a262acc1d9 tarantool]# cat ~/.gdbinit
set disable-randomization off

[root@b9a262acc1d9 tarantool]# gdb src/tarantool
<...>
(gdb) b main
Breakpoint 1 at 0x41c7b0: file /tarantool/src/main.cc, line 630.
(gdb) run
Starting program: /tarantool/src/tarantool 
Missing separate debuginfos, use: <...>
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib64/libthread_db.so.1".

Breakpoint 1, main (argc=1, argv=0x7ffc77daff98) at /tarantool/src/main.cc:630
630	{
Missing separate debuginfos, use: <...>
(gdb) p main
$1 = {int (int, char **)} 0x41c7b0 <main(int, char**)>
(gdb) p $rip
$2 = (void (*)(void)) 0x41c7b0 <main(int, char**)>
(gdb) quit

[root@b9a262acc1d9 tarantool]# gdb src/tarantool
<...>
(gdb) b main
Breakpoint 1 at 0x41c7b0: file /tarantool/src/main.cc, line 630.
(gdb) run
Starting program: /tarantool/src/tarantool 
Missing separate debuginfos, use: <...>
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib64/libthread_db.so.1".

Breakpoint 1, main (argc=1, argv=0x7fffdc531c08) at /tarantool/src/main.cc:630
630	{
Missing separate debuginfos, use: <...>
(gdb) p main
$1 = {int (int, char **)} 0x41c7b0 <main(int, char**)>
(gdb) p $rip
$2 = (void (*)(void)) 0x41c7b0 <main(int, char**)>
```

Reply:
It turned out that some compilers to not pass the `-pie` flag to the linker. Fixed by passing it explicitly.

Reply:
The same experiment now gives different addresses. Nice!